### PR TITLE
Slice integration for DataFrame

### PR DESCRIPTION
**Pull request by Kriyszig**
This PR brings with it a way to retrieve data of DataFrame as Slice and assign a Slice back to DataFrame
cc/ @thewilsonator, sorry this PR came a little later than expected. Sorry about that :disappointed:

### CODE REVIEW

File: source/magpie/dataframe.d [hunk 1]
```diff
@@ -1375,6 +1385,47 @@ public:
 
         return grp;
     }
+
+    auto asSlice(Type, SliceKind kind)() @property
+    {
+        static if(__traits(isArithmetic, Type))
```
Comment (thewilsonator) [inline source/magpie/dataframe.d]: do
```
static if(__traits(isArithmetic, Type))
    alias AType = Type;
else 
    alias AType = string;
```

then the two branches are virtually identical.

File: source/magpie/dataframe.d [hunk 2]
```diff
@@ -989,6 +989,26 @@ public:
         }
     }
 
+    /// Assign Slice to DataFrame
+    void opAssign(T, SliceKind k)(Slice!(T*, 2, k) input)
+    {
+        assert(input.shape[0] <= rows && input.shape[1] <= cols, "Given Slice is larger than DataFrame");
+        static foreach(i; 0 .. RowType.length)
```
Comment (thewilsonator) [inline source/magpie/dataframe.d]: ditto.

**Comment by thewilsonator** [comment]
Looking very nice! No problem on the timing, your health is more important.

File: source/magpie/dataframe.d (hunk 1 above)
Comment (Kriyszig) [reply source/magpie/dataframe.d]: Completely overlooked `Alias` :man_facepalming: 
Addressed in a1b2c3a

File: source/magpie/dataframe.d (hunk 2 above)
Comment (Kriyszig) [reply source/magpie/dataframe.d]: Addressed in a1b2c3a

**Pull request merged**